draft-ietf-tls-esni.md

```diff
@@ -723,38 +716,56 @@ error code.
 
 ### Handling HelloRetryRequest {#client-hrr}
 
-As required in {{real-ech}}, clients offering ECH MUST ensure that all
-extensions or parameters that might change in response to receiving a
-HelloRetryRequest have the same values in ClientHelloInner and
-ClientHelloOuter. That is, if a HelloRetryRequest causes a parameter to be
-changed, the same change is applied to both ClientHelloInner and
-ClientHelloOuter. Applicable parameters include:
+If the server sends a HelloRetryRequest in response to the ClientHello, the
+client first checks for the presence of the "encrypted_client_hello" extension.
+The presence of this extension is an indication that the client-facing server
+accepted ECH and that the backend server sent a HelloRetryRequest. If absent,
+then the client presumes rejection and proceeds as if the message was sent by
+the client-facing server.
+
+The HelloRetryRequest constructed by the backend server is called the
+HelloRetryRequestInner message; and the HelloRetryRequest constructed by the
+client-facing server is called the HelloRetryRequestOuter message.
```

Comment:
This text needs some work because it implicitly assumes this nested structure.

Reply:
Can you elaborate? The goal of this paragraph is merely to name HRRInne and HRROuter. How would you rework this text?

Reply:
My point is that you don't tell us that there is an HRRInner and HRROuter. So you need to say that first.

Reply:
Gotcha. Done.